### PR TITLE
Add new BBE for ballerina/file module

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose
This PR is to add new BBE for ballerina/file module.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: examples/file/file.out [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# To run this sample, navigate to the directory that contains the
+# `.bal` file and issue the `ballerina run` command.
+$ ballerina run file.bal
+
```
Comment (anupama-pathirage) [inline examples/file/file.out]: Shouldn't we display the output of the example here?

File: examples/file/file.bal [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import ballerina/file;
+import ballerina/filepath;
+import ballerina/io;
+
+public function main() {
+
+    // Get the current directory path.
+    io:println("Current directory: " + file:getCurrentDirectory());  // e.g. “/home/john/work”
```
Comment (anupama-pathirage) [inline examples/file/file.bal]: Shall we move this comment at the end also to the upper line?

File: examples/file/file.bal [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    // Get metadata information of the file.
+    file:FileInfo|error fileInfoResults = file:getFileInfo("bar.txt");
+    if (fileInfoResults is file:FileInfo) {
+        io:println("File name: " + fileInfoResults.getName());
+        io:println("File size: " + fileInfoResults.getSize().toString());
+        io:println("Is directory: " + fileInfoResults.isDir().toString());
+        io:println("Modified at " + fileInfoResults.getLastModifiedTime().toString());
+
```
Comment (anupama-pathirage) [inline examples/file/file.bal]: Extra new line

File: examples/file/file.bal [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    // Rename(Move) file or directory to new path.
+    string newFilePath = checkpanic filepath:build("foo", "bar1.txt");
+    error? renameResults = file:rename("bar.txt", newFilePath);
+    if (renameResults is ()) {
+        io:println("bar.txt file is moved to new path " + newFilePath);
+    }
+
+    // Get default directory use for temporary files.
```
Comment (anupama-pathirage) [inline examples/file/file.bal]:
```suggestion
    // Get default directory used for temporary files.
```

File: stdlib/file/src/main/ballerina/src/file/Module.md [hunk 5]
```diff
@@ -25,13 +25,13 @@ public function main() {
     string|error results = file:createDir(“foo/bar”);
     
     // Create a directory with any non existence parents.
-    string|error results = file:createDir(“foo/bar”, parentDirs = true);
+    string|error results = file:createDir(“foo/bar”, true);
```
Comment (anupama-pathirage) [inline stdlib/file/src/main/ballerina/src/file/Module.md]: We should use some `io:println() lines to validate the results of each of these operations. They should appear in .out file

File: examples/file/file.out (hunk 1 above)
Comment (daneshk) [reply examples/file/file.out]: Output value depends on where the user executes the example. we can add sample output.

File: stdlib/file/src/main/ballerina/src/file/Module.md (hunk 5 above)
Comment (anupama-pathirage) [reply stdlib/file/src/main/ballerina/src/file/Module.md]: sorry this comment applies to .bal file.

File: examples/file/file.out (hunk 1 above)
Comment (daneshk) [reply examples/file/file.out]: Fixed in 0516bf0

File: examples/file/file.description [hunk 6]
```diff
@@ -0,0 +1 @@
+// The Ballerina File API contains functions to perform file system operations.
```
Comment (praneesha) [inline examples/file/file.description]:
```suggestion
// The Ballerina File API contains functions to perform file-system operations.
```

File: examples/file/file.out [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# To run this sample, navigate to the directory that contains the
+# `.bal` file and issue the `ballerina run` command.
```
Comment (praneesha) [inline examples/file/file.out]:
```suggestion
# `.bal` file and execute the `ballerina run` command.
```

File: examples/file/file.out [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+Created file path: /A/B/bar.txt
+File name: bar.txt
+File size: 0
+Is directory: false
+Modified at time=1567597039000 zone=id=Z offset=0
+bar.txt file exists: true
+bar.txt file is copied to new path foo/bar/bar.txt
+bar.txt file is moved to new path foo/bar1.txt
+Temporary directory: /var/folders/4z/xvlnvsp54ks9793d6k5_6bw40000gn/T/
+Remove file at foo/bar1.txt
```
Comment (praneesha) [inline examples/file/file.out]:
```suggestion
Remove file at foo/bar1.txt.
```

File: examples/file/file.out [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+Remove foo directory with all child elements.
```
Comment (praneesha) [inline examples/file/file.out]:
```suggestion
Remove the `foo` directory with all its child elements.
```

File: examples/file/file.bal [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    // Remove file or directory in specified file path.
+    error? removeResults = file:remove(newFilePath);
+    if (removeResults is ()) {
+        io:println("Remove file at " + newFilePath);
+    }
+
+    // Remove directory in specified file path with all children.
+    removeResults = file:remove("foo", true);
+    if (removeResults is ()) {
+        io:println("Remove foo directory with all child elements.");
```
Comment (praneesha) [inline examples/file/file.bal]:
```suggestion
        io:println("Remove foo directory with all its child elements.");
```

File: stdlib/file/src/main/ballerina/src/file/Module.md [hunk 11]
```diff
@@ -25,13 +25,13 @@ public function main() {
     string|error results = file:createDir(“foo/bar”);
     
     // Create a directory with any non existence parents.
```
Comment (praneesha) [inline stdlib/file/src/main/ballerina/src/file/Module.md]:
```suggestion
    // Create a directory with any none-existent parents.
```

File: stdlib/file/src/main/ballerina/src/file/Module.md [hunk 12]
```diff
@@ -25,13 +25,13 @@ public function main() {
...
-    string|error results = file:createDir(“foo/bar”, parentDirs = true);
+    string|error results = file:createDir(“foo/bar”, true);
     
     // Remove file or directory in specified file path.
```
Comment (praneesha) [inline stdlib/file/src/main/ballerina/src/file/Module.md]:
```suggestion
    // Remove the file or directory in the specified file path.
```

File: stdlib/file/src/main/ballerina/src/file/Module.md [hunk 13]
```diff
@@ -25,13 +25,13 @@ public function main() {
...
     error? results = file:remove(“foo/bar.txt”);
 
     // Remove directory in specified file path with all children.
```
Comment (praneesha) [inline stdlib/file/src/main/ballerina/src/file/Module.md]:
```suggestion
    // Remove the directory in the specified file path with all its children.
```

File: stdlib/file/src/main/ballerina/src/file/Module.md [hunk 14]
```diff
@@ -25,13 +25,13 @@ public function main() {
...
-    error? results = file:remove(“foo/bar”, recursive = true);
+    error? results = file:remove(“foo/bar”, true);
     
     // Rename(Move) file or directory to new path.
```
Comment (praneesha) [inline stdlib/file/src/main/ballerina/src/file/Module.md]:
```suggestion
    // Rename(Move) the file or directory to the new path.
```

File: stdlib/file/src/main/ballerina/src/file/Module.md [hunk 15]
```diff
@@ -43,12 +43,12 @@ public function main() {
     string|error results = file:createFile(“bar.txt”);
     
     // Get metadata information of the file.
-    FileInfo|error result = file:getFileInfo(“foo/bar.txt”);
+    file:FileInfo|error result = file:getFileInfo(“foo/bar.txt”);
     
     // Get list of files in the directory.
```
Comment (praneesha) [inline stdlib/file/src/main/ballerina/src/file/Module.md]:
```suggestion
    // Get the list of files in the directory.
```

File: stdlib/file/src/main/ballerina/src/file/Module.md [hunk 16]
```diff
@@ -43,12 +43,12 @@ public function main() {
...
-    FileInfo[]|error results = file:readDir(“foo/bar”);
+    file:FileInfo[]|error results = file:readDir(“foo/bar”);
     
     // Copy file ot directory to new path.
```
Comment (praneesha) [inline stdlib/file/src/main/ballerina/src/file/Module.md]:
```suggestion
    // Copy the file or directory to the new path.
```

**Pull request merged**